assets/js/main.js

```diff
@@ -0,0 +1,249 @@
+/*global
+  $
+  Audio
+*/
+
+(function () {
+  // ================== globals ==================
+  // keycodes for left and right
+  var LEFT = 37;
+  var RIGHT = 39;
+
+  // size of each grid square
+  var GRID_SIZE = 12;
+  // grid height and width
+  var GRID_WIDTH = 50;
+  var GRID_HEIGHT = 40;
+
+  // initial game settings
+  // how fast the game update loop runs
+  var START_TICK = 10;
+  // how many update calls per drop move / collision
+  var START_DROPS_SPEED = 10;
+  // how many update calls per pad movement
+  var START_MOVE_SPEED = 5;
+  // how widet he pad is to start
+  var START_PAD_WIDTH = 4;
+  // how many update calls per createDrop
+  var START_DROP_CREATE_TIME = 99;
+
+  // the jquery game element
+  var $game = null;
+  // the pad that catches the drops
+  var $pad = null;
+  // the jquery score element
+  var $score = null;
+  // all the jquery drop elements
+  var $drops = [];
+
+  // all the game state
+  var game = null;
+
+  // the sound of catching
+  var catchSound = null;
+
+  // ================== utils ==================
+
+  // play an Audio object restarting if sound currently playing
+  // todo overlapping sounds we must create new Audio instances
+  var playSound = function (sound) {
+    // if a sound is currently playing restart from beginning
+    if (!sound.paused) {
+      sound.currentTime = 0;
+    } else {
+      sound.play();
+    }
+  };
+
+  // ================== view helpers ==================
+  // these are all helpers to get and set model info from a jquery element
+  // the conversion to grid is also handled here
```

Comment:
This is an area where the students will probably do it different / struggle.  I can do another version where the GRID and view logic is scattered throughout the code.  If we can encourage them to do this it does make it much easier to code.